src/app/modules/shared_models/members_model.nim

```diff
@@ -0,0 +1,109 @@
+import NimQml, Tables, strformat
+import  ./member_item
+import ../../global/global_singleton
+
+type
+  MembersRoles {.pure.} = enum
+    PubKey = UserRole + 1
+    # LastSeen = UserRole,
+    # StatusType = UserRole,
+    # Online = UserRole,
+    # SortKey = UserRole
+
+QtObject:
+  type
+    MembersModel* = ref object of QAbstractListModel
+      members*: seq[MemberItem]
+
+  proc setup(self: MembersModel) = self.QAbstractListModel.setup
+
+  proc delete(self: MembersModel) =
+    self.QAbstractListModel.delete
+
+  proc newMembersModel*(members: seq[MemberItem]): MembersModel =
+    new(result, delete)
+    result.members = members
+    result.setup()
+
+  proc `$`*(self: MembersModel): string =
+    for i in 0 ..< self.members.len:
+      result &= fmt"""MembersModel:
+      [{i}]:({$self.members[i]})
+      """
+
+  proc getIndexFromPubKey*(self: MembersModel, pubKey: string): int =
+    var i = 0
+    for member in self.members:
+      if (member.id == pubKey):
+        return i
+      i = i + 1
+    return -1
+
+  proc hasMember*(self: MembersModel, pubkey: string): bool =
+    for member in self.members:
+      if (member.id == pubkey):
+        return true
+    return false
+
+  proc removeMember*(self: MembersModel, pubKey: string) =
+    let memberIndex = self.getIndexFromPubKey(pubKey)
+    if (memberIndex == -1):
+      return
+    self.beginRemoveRows(newQModelIndex(), memberIndex, memberIndex)
+    self.members.delete(memberIndex)
+    self.endRemoveRows()
+
+  proc countChanged*(self: MembersModel) {.signal.}
+  proc count*(self: MembersModel): int {.slot.} =
+    self.members.len
+
+  QtProperty[int] count:
+    read = count
+    notify = countChanged
+
+  method rowCount(self: MembersModel, index: QModelIndex = nil): int = 
+    self.members.len
+
+  # proc memberStatus(self: MembersModel, pk: string): int =
```

Comment:
All these comments are for the later stage of making the member list also show the status of who's online